tests/accessor/accessor_common.h

```diff
@@ -374,32 +376,54 @@ void check_def_constructor(GetAccFunctorT get_accessor_functor) {
   auto queue = once_per_unit::get_queue();
   sycl::range<1> r(1);
   const size_t conditions_checks_size = 8;
-  bool conditions_check[conditions_checks_size]{false};
+  bool conditions_check[conditions_checks_size];
+  std::fill(conditions_check, conditions_check + conditions_checks_size, true);
 
+  auto acc = get_accessor_functor();
   if constexpr (AccType != accessor_type::host_accessor) {
-    sycl::buffer res_buf(conditions_check, sycl::range(conditions_checks_size));
+    check_empty_accessor_constructor_post_conditions(acc, conditions_check,
+                                                     false);
+  } else {
+    check_empty_accessor_constructor_post_conditions(acc, conditions_check,
+                                                     true);
+  }
 
-    queue
-        .submit([&](sycl::handler& cgh) {
-          sycl::accessor res_acc(res_buf, cgh);
-          auto acc = get_accessor_functor();
-          if (acc.is_placeholder()) {
-            cgh.require(acc);
-          }
-          if constexpr (Target == sycl::target::host_task) {
-            cgh.host_task([=] {
-              check_empty_accessor_constructor_post_conditions(acc, res_acc);
-            });
-          } else if constexpr (Target == sycl::target::device) {
-            cgh.parallel_for_work_group(r, [=](sycl::group<1>) {
-              check_empty_accessor_constructor_post_conditions(acc, res_acc);
-            });
-          }
-        })
-        .wait_and_throw();
+  for (size_t i = 0; i < conditions_checks_size; i++) {
+    CHECK(conditions_check[i]);
+  }
+}
+
+/**
+ * @brief Common function that constructs placeholder accessor with zero-length
+ * buffer and checks post-conditions
+ *
+ * @tparam AccType Type of the accessor
+ * @tparam DataT Type of underlying data
+ * @tparam Dimension Dimensions of the accessor
+ * @tparam AccessMode Access mode of the accessor
+ * @tparam Target Target of accessor
+ * @tparam GetAccFunctorT Type of functor for accessor creation
+ */
+template <accessor_type AccType, typename DataT, int Dimension,
+          sycl::access_mode AccessMode = sycl::access_mode::read_write,
+          sycl::target Target = sycl::target::device, typename GetAccFunctorT>
+void check_zero_length_buffer_placeholder_constructor(
+    GetAccFunctorT get_accessor_functor) {
+  auto queue = once_per_unit::get_queue();
+  sycl::range<Dimension> r =
+      util::get_cts_object::range<Dimension>::get(0, 0, 0);
+  sycl::buffer<DataT, Dimension> data_buf(r);
+  const size_t conditions_checks_size = 8;
+  bool conditions_check[conditions_checks_size];
+  std::fill(conditions_check, conditions_check + conditions_checks_size, true);
+
+  auto acc = get_accessor_functor(data_buf);
+  if constexpr (AccType != accessor_type::host_accessor) {
```

Comment:
Why is there a difference for iterators between host and device?

Reply:
Could you summarize this in some comments around the code? We might forget about the rationale... :-)

Reply:
Comments added.